Data/Binary/Serialise/CBOR/Read.hs

```diff
@@ -1177,7 +1177,11 @@ tryConsumeDouble :: Word8 -> ByteString -> DecodedToken Double
 tryConsumeDouble hdr !bs = case fromIntegral hdr :: Word of
   0xf9 -> DecodedToken 3 (float2Double $ wordToFloat16 $ withBsPtr grabWord16 (BS.unsafeTail bs))
   0xfa -> DecodedToken 5 (float2Double $ wordToFloat32 $ withBsPtr grabWord32 (BS.unsafeTail bs))
+#if !defined(ghcjs_HOST_OS)
   0xfb -> DecodedToken 9                (wordToFloat64 $ withBsPtr grabWord64 (BS.unsafeTail bs))
+#else
+  0xfb -> DecodedToken 9                (float2Double $ wordToFloat32 $ withBsPtr grabWord32 (BS.unsafeTail bs))
```

Comment:
Unfortunately this is not going to work. We have to read the full 8 bytes. Reading the first 4 will just give us nonsense.

We'll need to work out how to read an IEEE double from 8 bytes on GHCJS.